src/iscsi.c

```diff
@@ -2075,7 +2084,12 @@ isboot_scsi_io(struct cam_sim *sim, union ccb *ccb)
 		}
 		pdu.ahs_addr = isboot_malloc_pdubuf(pdu.ahs_size);
 		if (pdu.ahs_addr == NULL) {
+			ISBOOT_ERROR("isboot_malloc_pdubuf out of memory\n");
+			ccb->ccb_h.status = CAM_REQ_CMP_ERR;
```

Comment:
I'd use CAM_BUSY here. It is retries after 500ms delay.  Generally allocation in I/O path is not good.